src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -192,12 +194,13 @@ internal override bool IsExpressionBodied
             MethodKind methodKind,
             bool isAutoPropertyAccessor,
             DiagnosticBag diagnostics) :
-            base(containingType, syntax.GetReference(), syntax.Body?.GetReference(), location)
+            base(containingType, syntax.GetReference(), syntax.Body?.GetReference() ?? syntax.ExpressionBody?.GetReference(), location)
         {
             _property = property;
             _explicitInterfaceImplementations = explicitInterfaceImplementations;
             _name = name;
             _isAutoPropertyAccessor = isAutoPropertyAccessor;
+            _isExpressionBodied = (syntax.Body == null && syntax.ExpressionBody != null) ? true : _property.IsExpressionBodied;
```

Comment:
In case `AccessorDeclarationSyntax syntax` has an expression body it makes sense to put `true` in the corresponding field. In all other cases it falls back to `_property.IsExpressionBodied`, what is the original behaviour before this feature was added. I understand that it doesn't make any sense for a `set` accessor, but I'm just 'reverting' to the old logic in that case.